tools/cosmovisor/cmd/cosmovisor/add_upgrade.go

```diff
@@ -25,13 +25,19 @@ func NewAddUpgradeCmd() *cobra.Command {
 
 	addUpgrade.Flags().Bool(cosmovisor.FlagForce, false, "overwrite existing upgrade binary / upgrade-info.json file")
 	addUpgrade.Flags().Int64(cosmovisor.FlagUpgradeHeight, 0, "define a height at which to upgrade the binary automatically (without governance proposal)")
+	addUpgrade.Flags().String(cosmovisor.FlagConfig, "", "path to cosmovisor config file")
 
 	return addUpgrade
 }
 
 // AddUpgrade adds upgrade info to manifest
 func AddUpgrade(cmd *cobra.Command, args []string) error {
-	cfg, err := cosmovisor.GetConfigFromEnv()
+	configPath, err := cmd.Flags().GetString(cosmovisor.FlagConfig)
```

Comment:
shouldn't we read the config in the root command and pass it down to sub commands?
this way we can have a persistent flag on the root command and avoid adding a flag to each command.

Reply:
Yes, I don't think we should allow to export during init. Imho init should always create the file (maybe ask for a confirmation if the file already exists).

Additionally for people upgrading having an export command that converts flags to a config flag would be handy.

Additionally this way you don't have clashing flag names as you just pointed out.

Reply:
Make sense.
So basically init command should only write to default path(leaving file already exist case aside).

additionally export command can read the config from the file(override if env is set) and print it.

Reply:
yes, sounds good!

Reply:
Done